### ISSUE TITLE
farm.data_handler.split_file encoding error

### ISSUE THREAD

**Issue by lingsond** [deepset-ai/FARM#462]
When I tried to use the split_file as in your train from scratch example, I get this error:
```
Splitting file ...:   5%|5         | 127877/2407713 [00:00<00:02, 869200.61it/s]
Traceback (most recent call last):
  File "finetune_lm.py", line 43, in <module>
    split_file(data_dir / "train.txt", output_dir=Path('/data/german_old_texts/processed/lm/split_files'), docs_per_file=20)
  File "/home/user/farm/data_handler/utils.py", line 785, in split_file
    write_file.writelines(lines_to_write)
UnicodeEncodeError: 'ascii' codec can't encode character '\xe4' in position 62: ordinal not in range(128)
```
From the split_file method definition in the source code, the default encoding should be utf-8. So I think this error shouldn't happened.

After reading the source code of this method, I noticed that the encoding is only used during open read the file. However, when opening the file for write (in line 784 and 793), you are using:

`write_file = stack.enter_context(open(filename, 'w+', buffering=10 * 1024 * 1024))` 

instead of

`write_file = stack.enter_context(open(filename, 'w+', encoding=encoding, buffering=10 * 1024 * 1024))`

Maybe this caused the error.

I don't know how I can change the source and test it to contribute.

Farm version = 0.4.6

**Comment by lingsond** [comment]
I added the encoding to the code in my fork. However, I'm doing the changes all on github, not on my local computer (I'm using pycharm as my ide, and doing git command manually has causing conflicts in the past, so I don't have any clone locally in my computer).
I did a PR to contribute, but this is my first time ever doing this, so I hope I'm doing everything right.

**Comment by Timoeller** [comment]
Hey, I have seen the PR and it looks good code wise. Thanks a lot.

We of course need to test if it works before merging.
How about you send me an email and we setup a short call so I can help you test it - I think that will be the fastest.

**Comment by lingsond** [comment]
@Timoeller , sorry, but I've been trying to look for a way to send you a private message/email, but I can't find a way to do that here on github. I don't find any information either on your profile page. Or should I do it via the contact form on deepset.ai website?

**Comment by Timoeller** [comment]
timo.moeller -at- deepset.ai

**Comment by lingsond** [comment]
@Timoeller , thank you again for taking your time to explain to me how to do things.

I've tested the revision locally now and it works as predicted. The error was indeed caused by the lack of adding the encoding during open write file. I created a new PR (after closing the old one. I don't know if this is needed. The old PR would pull from a branch that has been merged with master, whereas the new PR will pull from master). However, I noticed something else during the process, which I will create a new issue for that.

For cases like this, where I report a bug, as well as create a PR with the solution for the bug, should I then close this issue after submitting the PR? Or should I just leave the issue like this and let you guys close this issue?

**Comment by Timoeller** [comment]
Hey lingsond, no worries, always happy to help.

> The old PR would pull from a branch that has been merged with master, whereas the new PR will pull from master
If there are no conflicts github is taking care of it automatically. If there are merge conflicts the github UI will also show it. Then you should merge the latest master into your fork (that way you dont have to create a new PR but just update the old one).

And in the PR you can link issues with the hashtag. Just write "This PR fixes #<issue_number>" and it will automatically link and close issues once the PR is merged
